test/fail_compilation/retscope2.d

```diff
@@ -162,3 +162,25 @@ void foo800()
     scope Exception e;
     throw e;
 }
+
+/*************************************************/
+
+/*
+TEST_OUTPUT:
+---
+fail_compilation/retscope2.d(907): Error: address of variable `this` assigned to `p17568` with longer lifetime
+---
+*/
+
+#line 900
+
+int* p17568;
+struct T17568
+{
+    int a;
+    void escape() @safe scope
+    {
+        p17568 = &a;
+    }
+}
```

Comment:
It does compile, but should not. Filed as https://issues.dlang.org/show_bug.cgi?id=17725